src/context.rs

```diff
@@ -1260,6 +1312,8 @@ pub struct BlockContext {
   pub preskip_segid: bool,
   above_partition_context: Vec<u8>,
```

Comment:
the following fields look all similar, would make sense having something like the following?
``` rust
struct EdgeContext {
  above: Vec<u8>,
  left: [u8; MAX_MIB_SIZE],
}
```

Reply:
You mean, factor out of BlockContext struct as separate stuct?
I am no sure. These new contexts are part of av1 spec like the existing *_coeff_context and partition_context, that are used to choose the CDFs, thus should be saved and restored to encode/decode symbols correctly.